jinahub/rankers/MinRanker/minranker.py

```diff
@@ -0,0 +1,56 @@
+__copyright__ = "Copyright (c) 2020-2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from itertools import groupby
+from typing import List, Dict
+
+from jina import Executor, requests, DocumentArray
+from jina.logging.logger import JinaLogger
+
+
+class MinRanker(Executor):
+    """
+    :class:`MinRanker` aggregates the score of the matched doc from the matched chunks.
+    For each matched doc, the score is aggregated from all the matched chunks belonging to that doc.
+    
+    :param metric: the distance metric used in `scores`
+    :param default_traversal_paths: traverse path on docs, e.g. ['r'], ['c']
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+
+    def __init__(
+        self,
+        metric: str = None,
+        default_traversal_paths: List[str] = None,
+        *args,
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        self.logger = JinaLogger(self.__class__.__name__)
+        if not metric:
+            self.logger.error("metric should not be None, please set it the same as your indexer's metric")
```

Comment:
What I mean is that this checking is not necessary at all. If an argument to a function is required, simply do not provide a default value, i.e.

```python
def __init__(self, metric: str):
```
instead of 
```python
def __init__(self, metric: str = None):
```
This way python will throw a very clear error that you need to provide the `metric` argument. There is no need for us to implement it if it is already part of base Python.

Reply:
It's not sufficient only to say that we need this param, we should tell users how to set it in logging, I just fixed it. Can you check if that is ok?

Reply:
That's my fault not making it clear at first

Reply:
I still don't think this is the way to go. This description of what the metric value should be belongs in the parameter description, not in a warning that appears when the value is not set.

If your concern is that users will set an invalid value, then you would need some kind of validation. But validation  and missing args/defaults are two different things.